app/scripts/controllers/detect-tokens.js

```diff
@@ -112,7 +112,9 @@ export default class DetectTokensController {
       : tokenList;
 
     const tokensToDetect = [];
-    this.web3.setProvider(this._network._provider);
+    this.ethersProvider = new ethers.providers.Web3Provider(
```

Comment:
this also appears to be redundant since `this.ethersProvider` is updated whenever `this._network` is updated...

Reply:
Hi @adonesky1 , you are probably right. I just wanted to make sure all logic from web3 implementation is there. 
I think we can remove those 2 lines.

Reply:
So I will double check but I think I'm actually wrong on this one per my comment [here](https://github.com/MetaMask/metamask-extension/pull/15386#discussion_r949379852)

Reply:
Wait now I'm confused because I'm not seeing how `this.ethersProvider` is used at all on this controller...?

Reply:
@amerkadicE do you see something I'm missing?

Reply:
Looks like it should have been removed in this [PR](https://github.com/MetaMask/metamask-extension/commit/6e5c2f03bfbc4a0c8fa334b7beedb19cf35c9d73#diff-323d0cf46498be3850b971474905354ea5ccf7fa13745ad1e6eba59c5b586830L162)

Reply:
I think its safe to remove `this.ethersProvider` from this controller

Reply:
Done